nsls2tools/csx1/devices/lakeshore.py

```diff
@@ -0,0 +1,195 @@
+        self._timeout = timeout
+        self._lagtime = 120
+
+        # the channel to watch to see if we are done
+        self._target_channel = target
+
+        # parameters for done testing
+        self.mean_thresh = .01
+        self.ptp_thresh = .1
+
+    def _value_cb(self, value, timestamp, **kwargs):
+        self._cache.append((value, timestamp))
+
+        if (timestamp - self._cache[0][1]) < self._lagtime / 2:
+            return
+
+        while (timestamp - self._cache[0][1]) > self._lagtime:
+            self._cache.popleft()
+
+        buff = np.array([v[0] for v in self._cache])
+        if self._done_test(self._setpoint, buff):
+            self._done_sts._finished()
+            self._reset()
+
+    def _setpoint_cb(value, **kwargs):
+        print('in cb', value)
+        if value == self._setpoint:
+            self._done_sts._finished()
+            self.setpoint.clear_sub(self._setpoint_cb, 'value')
+
+    def _reset(self):
+        if self._target_channel == 'setpoint':
+            target = self.setpoint
+            target.clear_sub(self._setpoint_cb, 'value')
+        else:
+            target = getattr(self, self._target_channel).T
+            target.clear_sub(self._value_cb, 'value')
+        self._done_sts = None
+        self._setpoint = None
+        self._cache.clear()
+
+    def _done_test(self, target, buff):
+        mn = np.mean(np.abs(buff - target))
+
+        if mn > self.mean_thresh:
+            return False
+
+        if np.ptp(buff) > self.ptp_thresh:
+            return False
+
+        return True
+
+
+    def set(self, new_position, *, timeout=None):
+        # to be subscribed to 'value' cb on readback
+        sts = self._done_sts = DeviceStatus(self, timeout=timeout)
+        if self.setpoint.get() == new_position:
+            self._done_sts._finished()
+            self._done_sts = None
+            return sts
+
+        self._setpoint = new_position
+
+        self.setpoint.set(self._setpoint)
+
+        # todo, set up subscription forwarding
+        if self._target_channel == 'setpoint':
+            self.setpoint.subscribe(local_cb, 'value')
+        else:
+            target = getattr(self, self._target_channel).T
+            target.subscribe(self._value_cb, 'value')
+
+        return self._done_sts
+
+
+class DelayGeneratorChan(EpicsSignal):
+    def __init__(self, prefix, **kwargs):
+        super().__init__(prefix + '-RB', write_pv=prefix + '-SP', **kwargs)
+
+
+class DelayGenerator(Device):
+    A = Cpt(DelayGeneratorChan, '-Chan:A}DO:Dly')
+    B = Cpt(DelayGeneratorChan, '-Chan:B}DO:Dly')
+    C = Cpt(DelayGeneratorChan, '-Chan:C}DO:Dly')
+    D = Cpt(DelayGeneratorChan, '-Chan:D}DO:Dly')
+    E = Cpt(DelayGeneratorChan, '-Chan:E}DO:Dly')
+    F = Cpt(DelayGeneratorChan, '-Chan:F}DO:Dly')
+    G = Cpt(DelayGeneratorChan, '-Chan:G}DO:Dly')
+    H = Cpt(DelayGeneratorChan, '-Chan:H}DO:Dly')
+
+# Current-Voltage meter, driven in current mode
```

Comment:
In principle, we should not need a beter ioc.  It was commented out when we stopped using it.  (last used a year ago July).  With all of the software changes, I am not sure that the current IOC will work with this current bluesky device.  It is something I will test in the next days when there is no beam.

I think it was commented out because this information was not wished to be recorded or to prevent people from trying to use.  @danielballan What do you suggest for handling these kings of items (not regularly used).  I imagine that once we get a sample holder for this capability that allows us to use the loadlock, that this device will become used more often, but still not for every (or likely most) experiments.